porch/controllers/packagevariantsets/pkg/controllers/packagevariantset/packagevariantset_controller.go

```diff
@@ -291,11 +306,65 @@ func (r *PackageVariantSetReconciler) repositorySet(
 	return result, nil
 }
 
-func (r *PackageVariantSetReconciler) objectSet(ctx context.Context,
-	target *api.Target,
-	upstreamPackageName string) ([]*pkgvarapi.Downstream, error) {
-	// TODO: Implement this
-	return nil, fmt.Errorf("specifying a set of objects in the target is not yet supported")
+func (r *PackageVariantSetReconciler) objectSet(target *api.Target,
+	upstreamPackageName string,
+	selectedObjects map[resid.ResId]*yaml.RNode) ([]*pkgvarapi.Downstream, error) {
+	var result []*pkgvarapi.Downstream
+	for _, obj := range selectedObjects {
+		downstreamPackageName, err := r.getDownstreamPackageName(target.PackageName,
+			upstreamPackageName, obj)
+		if err != nil {
+			return nil, err
+		}
+		repo, err := r.fetchValue(target.Objects.RepoName, obj)
+		if err != nil {
+			return nil, err
+		}
+		if repo == "" {
+			return nil, fmt.Errorf("error evaluating repo name: received empty string")
+		}
+		result = append(result, &pkgvarapi.Downstream{
+			Package: downstreamPackageName,
+			Repo:    repo,
+		})
+	}
+	return result, nil
+}
+
+func (r *PackageVariantSetReconciler) getSelectedObjects(ctx context.Context, selectors []api.Selector) (map[resid.ResId]*yaml.RNode, error) {
+	selectedObjects := make(map[resid.ResId]*yaml.RNode) // this is a map to prevent duplicates
+
+	for _, selector := range selectors {
+		uList := &unstructured.UnstructuredList{}
+		group, version := resid.ParseGroupVersion(selector.APIVersion)
+		uList.SetGroupVersionKind(schema.GroupVersionKind{
+			Group:   group,
+			Version: version,
+			Kind:    selector.Kind,
+		})
+
+		labelSelector, err := metav1.LabelSelectorAsSelector(selector.Labels)
+		if err != nil {
+			return nil, err
+		}
+
+		if err := r.Client.List(ctx, uList,
```

Comment:
I think there are errors that could come here where we should probably expose an error in the status rather than retry. If the provided GVK is not a known type would be one example.

Reply:
done. Also moved validation errors to status.